web/src/main/webapp/xslt/services/metadata/validate-fn.xsl

```diff
@@ -97,7 +97,12 @@
       <!--cvc-type.3.1.3: The value 'DUMMY_DENOMINATOR' of element 'gco:Integer' is not valid. (Element: gco:Integer with parent element: gmd:denominator)-->
       <rule errorType="type.3.1.3withparent">The value '(.*)' of element '(.*)' is not valid\. \(Element: ([a-z]{3}):(.*) with parent element: (.*)\)</rule>
       <rule errorType="type.3.1.3">The value '(.*)' of element '(.*)' is not valid\.</rule>
-      <rule errorType="enumeration-valid">Value '(.*)' is not facet-valid with respect to enumeration '\[(.*)\]'\. It must be a value from the enumeration\. \(Element: ([a-z]{3}):(.*) with parent element: (.*)\)</rule>
+      <rule errorType="enumeration-validwithparent">Value '(.*)' is not facet-valid with respect to enumeration '\[(.*)\]'\. It must be a value from the enumeration\. \(Element: ([a-z]{3}):(.*) with parent element: (.*)\)</rule>
+      <rule errorType="enumeration-valid">Value '(.*)' is not facet-valid with respect to enumeration '\[(.*)\]'\. It must be a value from the enumeration\.</rule>
+      <!--datatype-valid.1.2.3: '' is not a valid value of union type 'Date_Type'.-->
+      <rule errorType="datatype-valid.1.2.3withparent">'(.*)' is not a valid value of union type '(.*)'\. \(Element: ([a-z]{3}):(.*) with parent element:(.*)\)</rule>
+      <rule errorType="datatype-valid.1.2.3">'(.*)' is not a valid value of union type '(.*)'\.</rule>
+
     </xsl:variable>
```

Comment:
I put test code like 

<xsl:otherwise>
	<xsl:message>Skiped $error=<xsl:value-of select="$error"></xsl:value-of></xsl:message>
       <xsl:value-of select="$error"/>
</xsl:otherwise>

So I summarized all skipped patterns:
Skiped $error='' is not a valid value of union type 'Date_Type'.
Skiped $error='' is not a valid value of union type 'Date_Type'.
Skiped $error='' is not a valid value of union type 'Date_Type'. (Element: gco:Date with parent element: gmd:date)
Skiped $error='' is not a valid value of union type 'Date_Type'. (Element: gco:Date with parent element: gmd:date)
Skiped $error=Value '' is not facet-valid with respect to enumeration '[farming, biota, boundaries, climatologyMeteorologyAtmosphere, economy, elevation, environment, geoscientificInformation, health, imageryBaseMapsEarthCover, intelligenceMilitary, inlandWaters, location, oceans, planningCadastre, society, structure, transportation, utilitiesCommunication]'. It must be a value from the enumeration. (Element: gmd:MD_TopicCategoryCode with parent element: gmd:topicCategory)


Realized there are three uncaught errors: enumeration-validwithparent, datatype-valid.1.2.3withparent, datatype-valid.1.2.3. So I added them accordingly.